src/loaddb/load_server_loader.cpp

```diff
@@ -124,6 +134,11 @@ namespace cubload
     cubthread::entry &thread_ref = cubthread::get_entry ();
     bool is_syntax_check_only = m_session.get_args ().syntax_check;
 
+    char *lower_case_class_name = (char *) db_private_alloc (NULL, intl_identifier_lower_string_size (class_name) + 1);
```

Comment:
please a cubmem unique_pointer or extensible (maybe stack) buffer to automatically manage memory deallocation on destruction.

since code is repeated many times, a function does all necessary steps - memory allocation and copying lowercase string - should be useful.